include/mbedtls/platform_util.h

```diff
@@ -200,6 +200,51 @@ struct tm *mbedtls_platform_gmtime_r( const mbedtls_time_t *tt,
                                       struct tm *tm_buf );
 #endif /* MBEDTLS_HAVE_TIME_DATE */
 
+/**
+ * \brief Write data into buffer cache.
+ *
+ * \param[in] context   contexts pointer used for buffer management.
+ * \param[in] buf       buffer that will hold the data
+ * \param[in] len       how many bytes must be written
+ *
+ * \return         The (positive) number of bytes write if successful.
+ *
+ * \note           An alternative implement can put its context pointer in
+ *                 context.
+ *
+ * \note           For time being, it is design for TLS 1.3 early data.
+ *
+ * \note           Before mbedtls_ssl_read_early_data() is called,
+ *                 an intermediate buffer is needed to reserve 0-RTT data. And
+ *                 it is difficult to guarantee that data will be read
+ *                 immediately. The buffer cache is to reserve 0-RTT data until
+ *                 it is read.
+ *
+ * \note           If MBEDTLS_PLATFORM_BUF_CACHE_ALT is defined, then Mbed TLS
+ *                 will unconditionally use the alternative implementation for
+ *                 mbedtls_platform_buf_cache_{write,read}() supplied by the user
+ *                 at compile time.
+ */
+MBEDTLS_CHECK_RETURN_CRITICAL
+int mbedtls_platform_buf_cache_write( void **context, const void *buf, size_t len );
+
+/**
+ * \brief Read data from buffer cache.
+ *
+ * \param[in] context   contexts used for buffer management.
+ * \param[in] buf       buffer that will hold the data.
+ * \param[in] len       maximum number of bytes to read
+ *
+ * \return         The (positive) number of bytes read if successful.
+ *
+ * \note        \b buf might be NULL, that means all buffers should be discarded.
+ *
+ * \note        See mbedtls_platform_buf_cache_write()
+ */
+MBEDTLS_CHECK_RETURN_CRITICAL
+int mbedtls_platform_buf_cache_read(
```

Comment:
Should we implement the cache buffer implicitly? I mean, maybe users have no need to know about them.

Reply:
That followup same logic as `MBEDTLS_PLATFORM_GMTIME_R_ALT` . If user want to change it, user can provide their own implementation. 

For our users, the implicit implementation might not meet their requirements, then user should provide them.